CrazyCanvas/Source/ECS/Systems/GUI/HUDSystem.cpp

```diff
@@ -223,13 +223,14 @@ bool HUDSystem::OnWeaponFired(const WeaponFiredEvent& event)
 		const ComponentArray<WeaponComponent>* pWeaponComponents = pECS->GetComponentArray<WeaponComponent>();
 		const ComponentArray<PlayerLocalComponent>* pPlayerLocalComponents = pECS->GetComponentArray<PlayerLocalComponent>();
 
-		for (Entity playerWeapon : m_WeaponEntities)
+		for (Entity weapon : m_WeaponEntities)
 		{
-			const WeaponComponent& weaponComponent = pWeaponComponents->GetConstData(playerWeapon);
+			const WeaponComponent& weaponComponent = pWeaponComponents->GetConstData(weapon);
 
-			if (pPlayerLocalComponents->HasComponent(weaponComponent.WeaponOwner) && m_HUDGUI)
+			if (weaponComponent.WeaponOwner == event.WeaponOwnerEntity && pPlayerLocalComponents->HasComponent(event.WeaponOwnerEntity))
 			{
 				m_HUDGUI->UpdateAmmo(weaponComponent.WeaponTypeAmmo, event.AmmoType);
+				LOG_ERROR("Entity %d fired", weaponComponent.WeaponOwner);
```

Comment:
This is not an error, right? Guessing it's an old debug thingy

Reply:
correct my old frend, mye dutti